Runtime/JSAsyncScope.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Threading;
+
+namespace NodeApi
+{
+    /// <summary
+    /// Keeps JSSynchronizationContext queue alive until the end of the 
+    public class JSAsyncScope : IDisposable
```

Comment:
Or can it be combined somehow with `JSValueScope`, which is already created and disposed for every callback?

Reply:
No, I do not see how we can combine them. They have different lifetime:
- `JSValueScope` is attached to the same call stack.
- `JSAsyncScope` is spans across multiple call stacks: we create it in one task, and then dispose it eventually in one of the continuation tasks. Its goal is to say to `JSSynchronizationContext` when we start using it and when we release it. 
If we would combine the two, then `JSValue` created in the initial call stack would not be able to understand that the underlying `napi_value` is already destroyed because we would fail to Dispose it at the right time.

Reply:
Make the JSAsyncScope a struct.